tests/integration-tests/tests/name_server_pool_tests.rs

```diff
@@ -262,97 +262,55 @@ fn test_local_mdns() {
-
-    let v4_record = v4_record(query.name().clone(), Ipv4Addr::new(127, 0, 0, 2));
-    let success_msg = message(query.clone(), vec![v4_record], vec![], vec![]);
-
-    let tcp_message = success_msg.clone();
-    let udp_message = success_msg;
+    let mut nx_message = message(query.clone(), vec![], vec![], vec![]);
+    nx_message.set_response_code(ResponseCode::NXDomain);
 
-    // fail the first udp request
-    let udp_nameserver = mock_nameserver_trust_nx(
-        vec![
-            Ok(udp_message.into()),
-            servfail_message.clone().map(Into::into),
-        ],
-        options,
-        false,
-    );
-    let tcp_nameserver = mock_nameserver_trust_nx(
-        vec![Err(ResolveError::from("Forced Testing Error"))],
-        options,
-        false,
+    let success_msg = message(
+        query.clone(),
+        vec![v4_record(query.name().clone(), Ipv4Addr::new(127, 0, 0, 2))],
+        vec![],
+        vec![],
     );
 
-    let mut pool = mock_nameserver_pool(vec![udp_nameserver], vec![tcp_nameserver], None, options);
+    // Fail the first UDP request.
+    let fail_nameserver =
+        mock_nameserver_trust_nx(vec![Ok(nx_message.into())], ResolverOpts::default(), true);
+    let succeed_nameserver =
+        mock_nameserver_trust_nx(vec![Ok(success_msg.into())], ResolverOpts::default(), true);
 
-    // lookup on UDP succeeds, any other would fail
-    let request = message(query.clone(), vec![], vec![], vec![]);
-    let future = pool.send(request).first_answer();
-
-    let response = block_on(future).unwrap();
-    assert!(response.response_code() == ResponseCode::ServFail);
-
-    // fail all udp succeed tcp
-    let udp_nameserver = mock_nameserver(vec![servfail_message.map(Into::into)], options);
-    let tcp_nameserver = mock_nameserver(vec![Ok(tcp_message.into())], options);
-
-    let mut pool = mock_nameserver_pool(vec![udp_nameserver], vec![tcp_nameserver], None, options);
+    let mut pool = mock_nameserver_pool(
+        vec![fail_nameserver, succeed_nameserver],
+        vec![],
+        None,
+        ResolverOpts::default(),
+    );
 
+    // Lookup on UDP should fail, since we trust nx responses.
+    // (If we retried the query with the second name server, we'd see a successful response.)
     let request = message(query, vec![], vec![], vec![]);
     let future = pool.send(request).first_answer();
-
-    let response = block_on(future).unwrap();
-    assert!(response.response_code() == ResponseCode::ServFail);
+    let response = block_on(future).expect_err("lookup request should fail with NXDOMAIN");
+    match response.kind() {
+        ResolveErrorKind::NoRecordsFound { response_code, .. }
+            if *response_code == ResponseCode::NXDomain => {}
+        kind => panic!(
+            "got unexpected kind of resolve error; expected `NoRecordsFound` error with NXDOMAIN,
+            got {:#?}",
+            kind,
+        ),
+    }
 }
 
 #[test]
 fn test_distrust_nx_responses() {
```

Comment:
Is this test totally invalid now?

Reply:
It's basically testing the exact same thing as the test below it, which i added in https://github.com/bluejekyll/trust-dns/pull/1526 — that when `trust_nx_responses` is set to `false`, we see the resolver try other name servers after getting an negative response.

So, I made sure that `SERVFAIL`, which is tested in `test_distrust_nx_responses`, was covered in the test below, and combined them. Does that make sense?